module Result {
  fun error (input : a) : Result(a, b) {
    `new Err(#{input})`
  }
}

component Main {
  fun render : String {
    try {
      x =
        Result.error("Blah")

      "Hello"
    } catch {
      "Blah"
    }
  }
}
--------------------------------------------------------------------------------
const B = new(class extends _M {
  b(c) {
    return new Err(c)
  }
})

class A extends _C {
  render() {
    return (() => {
      let _catch_all = () => {
        return `Blah`
      }

      let _0 = B.b(`Blah`)

      if (_0 instanceof Err) {
        let _error = _0.value
        return _catch_all()
      }

      let a = _0.value

      return `Hello`
    })()
  }
}

A.displayName = "Main"
